src/Core/ServerSettings.h

```diff
@@ -76,8 +76,9 @@ namespace DB
     M(Double, index_mark_cache_size_ratio, DEFAULT_INDEX_MARK_CACHE_SIZE_RATIO, "The size of the protected queue in the secondary index mark cache relative to the cache's total size.", 0) \
     M(UInt64, mmap_cache_size, DEFAULT_MMAP_CACHE_MAX_SIZE, "A cache for mmapped files.", 0) \
     \
-    M(Bool, disable_internal_dns_cache, false, "Disable internal DNS caching at all.", 0) \
-    M(Int32, dns_cache_update_period, 15, "Internal DNS cache update period in seconds.", 0) \
+    M(Bool,   disable_internal_dns_cache, false, "Disable internal DNS caching at all.", 0) \
+    M(UInt64, dns_cache_max_size, 1024, "Internal DNS cache max size in bytes.", 0) \
```

Comment:
1024 bytes seems like a ridiculously small cache size. We are not talking gigabytes here, let's increase to 1 MB.

Reply:
good idea, i will submit PR soon

Reply:
https://github.com/ClickHouse/ClickHouse/pull/60500